src/lib/common/sem.cpp

```diff
@@ -294,55 +294,86 @@ int transSemGive(const char* who, const char* what)
 static pthread_mutex_t contexts_mutex = PTHREAD_MUTEX_INITIALIZER;
 static std::map<std::string, struct curl_context> contexts;
 
-//Statistics
+// Statistics
 static struct timespec accCCMutexTime = { 0, 0 };
 
 
-int get_curl_context(const std::string& key, struct curl_context *pcc)
+/* ****************************************************************************
+*
+* curl_context_cleanup - 
+*/
+void curl_context_cleanup(void)
```

Comment:
Maybe we should create a tech debt issue about sem.cpp refactoring to put these functions out. In addition, probably a better name for sem.cpp/sem.h is locks.cpp/looks.h (given that not all locks are based on semaphores, some are based on pthread_mutex, if I'm remembering correctly).

I'd create the issue on Monday. If you have the opportunity to do it before, please go ahead.

NTC (in this PR).